src/plugins/discover/public/application/components/discover_grid/discover_grid_document_selection.tsx

```diff
@@ -0,0 +1,169 @@
+ */
+export const getDocId = (doc: ElasticSearchHit & { _routing?: string }) => {
+  const routing = doc._routing ? doc._routing : '';
+  return [doc._index, doc._id, routing].join('::');
+};
+export const SelectButton = ({ rowIndex }: { rowIndex: number }) => {
+  const ctx = useContext(DiscoverGridContext);
+  const doc = useMemo(() => ctx.rows[rowIndex], [ctx.rows, rowIndex]);
+  const id = useMemo(() => getDocId(doc), [doc]);
+  const checked = useMemo(() => ctx.selectedDocs.includes(id), [ctx.selectedDocs, id]);
+
+  return (
+    <EuiCheckbox
+      id={id}
+      label=""
+      checked={checked}
+      data-test-subj={`dscGridSelectDoc-${id}`}
+      onChange={() => {
+        if (checked) {
+          const newSelection = ctx.selectedDocs.filter((docId) => docId !== id);
+          ctx.setSelectedDocs(newSelection);
+        } else {
+          ctx.setSelectedDocs([...ctx.selectedDocs, id]);
+        }
+      }}
+    />
+  );
+};
+
+export function DiscoverGridDocumentToolbarBtn({
+  isFilterActive,
+  rows,
+  selectedDocs,
+  setIsFilterActive,
+  setSelectedDocs,
+}: {
+  isFilterActive: boolean;
+  rows: ElasticSearchHit[];
+  selectedDocs: string[];
+  setIsFilterActive: (value: boolean) => void;
+  setSelectedDocs: (value: string[]) => void;
+}) {
+  const [isSelectionPopoverOpen, setIsSelectionPopoverOpen] = useState(false);
+
+  const getMenuItems = useCallback(() => {
+    return [
+      isFilterActive ? (
+        <EuiContextMenuItem
+          data-test-subj="dscGridShowAllDocuments"
+          key="showAllDocuments"
+          icon="eye"
+          onClick={() => {
+            setIsSelectionPopoverOpen(false);
+            setIsFilterActive(false);
+          }}
+        >
+          <FormattedMessage id="discover.showAllDocuments" defaultMessage="Show all documents" />
+        </EuiContextMenuItem>
+      ) : (
+        <EuiContextMenuItem
+          data-test-subj="dscGridShowSelectedDocuments"
+          key="showSelectedDocuments"
+          icon="eye"
+          onClick={() => {
+            setIsSelectionPopoverOpen(false);
+            setIsFilterActive(true);
+          }}
+        >
+          <FormattedMessage
+            id="discover.showSelectedDocumentsOnly"
+            defaultMessage="Show selected documents only"
+          />
+        </EuiContextMenuItem>
+      ),
+
+      <EuiContextMenuItem
+        data-test-subj="dscGridClearSelectedDocuments"
+        key="clearSelection"
+        icon="cross"
+        onClick={() => {
+          setIsSelectionPopoverOpen(false);
+          setSelectedDocs([]);
+          setIsFilterActive(false);
+        }}
+      >
+        <FormattedMessage id="discover.clearSelection" defaultMessage="Clear selection" />
+      </EuiContextMenuItem>,
+      <EuiCopy
+        key="copyJsonWrapper"
+        data-test-subj="dscGridCopySelectedDocumentsJSON"
+        textToCopy={
+          !rows ? '' : JSON.stringify(rows.filter((row) => selectedDocs.includes(getDocId(row))))
```

Comment:
nit: I think reversing the order might improve readability and get rid of the negation:
`rows ? JSON.stringify() : ' ' `

Reply:
very true! thx!